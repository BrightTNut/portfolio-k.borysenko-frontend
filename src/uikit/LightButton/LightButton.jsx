import s from './LightButton.module.css';

const LightButton = ({ name, onClick, ariaLabel }) => {
  return (
    <button className={s.button} onClick={onClick} aria-label={ariaLabel}>
      {name}
    </button>
  );
};

export default LightButton;
