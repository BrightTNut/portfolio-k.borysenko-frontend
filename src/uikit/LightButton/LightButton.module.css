.button {
  max-width: 190px;
  font-size: var(--font-size-button);
  letter-spacing: 1.5px;
  border: 1px solid var(--text-primary);
  background-color: transparent;
  padding: 14px 0;
  width: 100%;
  transition: color 250ms cubic-bezier(0.4, 0, 0.2, 1),
    border-color 250ms cubic-bezier(0.4, 0, 0.2, 1);
}

.button:hover {
  cursor: pointer;
  color: var(--ui-hover);
  border-color: var(--ui-hover);
}

@media screen and (max-width: 414px) {
  .button {
    padding: 8px 0;
    font-size: smaller;
  }
}
