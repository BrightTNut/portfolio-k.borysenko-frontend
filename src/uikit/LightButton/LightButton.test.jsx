import React from 'react';
import { render, fireEvent, screen } from '@testing-library/react';
import '@testing-library/jest-dom';
import LightButton from './LightButton';

describe('LightButton Component', () => {
  it('renders correctly with given name', () => {
    render(<LightButton name="Test Button" />);
    expect(screen.getByText('Test Button')).toBeInTheDocument();
  });

  it('calls onClick prop when clicked', () => {
    const handleClick = jest.fn();
    render(<LightButton name="Clickable" onClick={handleClick} />);
    fireEvent.click(screen.getByText('Clickable'));
    expect(handleClick).toHaveBeenCalledTimes(1);
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<LightButton name="Snapshot Test" />);
    expect(asFragment()).toMatchSnapshot();
  });
});
