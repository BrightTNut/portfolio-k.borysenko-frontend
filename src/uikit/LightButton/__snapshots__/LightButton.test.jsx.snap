// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LightButton Component matches snapshot 1`] = `
<DocumentFragment>
  <button
    class="button"
  >
    Snapshot Test
  </button>
</DocumentFragment>
`;
